presto-docs/src/main/sphinx/security/server.rst

```diff
@@ -0,0 +1,254 @@
+JDBC and ODBC drivers.
+
+.. warning::
+
+  Worker nodes cannot yet be configured to connect to the Presto coordinator
+  using HTTPS or to authenticate with Kerberos. It is the administrator's
+  responsibility to enable unauthenticated access over HTTP for worker nodes
+  and ensure unathenticated access is blocked for any node that is not a worker
+  node. For nodes that are not worker nodes, block access to the Presto
+  coordinator's HTTP port.
+
+To enable Kerberos authentication for Presto, configuration changes are made on
+the Presto coordinator. No changes are required to the worker configuration;
+the worker nodes will continue to connect to the coordinator over
+unauthenticated HTTP.
+
+Environment Configuration
+-------------------------
+
+.. |subject_node| replace:: client
+.. include:: kerberos-services.fragment
+.. include:: kerberos-configuration.fragment
+
+Kerberos Principals and Keytab Files
+^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+The Presto coordinator needs a Kerberos principal, as do users who are going to
+connect to the Presto coordinator. You will need to create these users in
+Kerberos using `kadmin
+<http://web.mit.edu/kerberos/krb5-latest/doc/admin/admin_commands/kadmin_local.html>`_.
+
+In addition, the Presto coordinator needs a `keytab file
+<http://web.mit.edu/kerberos/krb5-devel/doc/basic/keytab_def.html>`_. After you create the principal, you can create the keytab file using :command:`kadmin`
+
+.. code-block:: none
+
+  kadmin
+  > addprinc -randkey presto@EXAMPLE.COM
+  > addprinc -randkey presto/presto-coordinator.example.com@EXAMPLE.COM
+  > ktadd -k /etc/presto/presto.keytab presto@EXAMPLE.COM
+  > ktadd -k /etc/presto/presto.keytab presto/presto-coordinator.example.com@EXAMPLE.COM
+
+.. include:: ktadd-note.fragment
+
+.. include:: jce-policy.fragment
+
+.. _coordinator_java_keystore:
+
+Java Keystore File for TLS
+^^^^^^^^^^^^^^^^^^^^^^^^^^
+
+Access to the Presto coordinator should be through https when using Kerberos
+authentication. The Presto coordinator needs keys to secure the :abbr:`TLS
+(Transport Layer Security)` connection. These keys are generated using
+:command:`keytool` and stored in a Java Keystore file for the Presto
+coordinator.
+
+.. code-block:: none
+  
+  keytool -genkeypair -alias presto -keyalg RSA -keystore keystore.jks
+  Enter keystore password:
+  Re-enter new password:
+  What is your first and last name?
+    [Unknown]:  presto-coordinator
+  What is the name of your organizational unit?
+    [Unknown]:
+  What is the name of your organization?
+    [Unknown]:
+  What is the name of your City or Locality?
+    [Unknown]:
+  What is the name of your State or Province?
+    [Unknown]:
+  What is the two-letter country code for this unit?
+    [Unknown]:
+  Is CN=eiger, OU=Unknown, O=Unknown, L=Unknown, ST=Unknown, C=Unknown correct?
+    [no]:  yes
+
+  Enter key password for <presto>
+          (RETURN if same as keystore password):  
+
```

Comment:
Suggest moving the next two paras before the code, and adding:

In the command line, the alias....

You'll be prompted for the first and last name. Use the Common Name that will be used in the certificate. In this case it should be the unqualified hostname of the Presto coordinator. In the example, you can see this in the prompt that confirms the information is correct.